Reexport scanl'

cocreature opened the pull request:
This is useful for the same reasons that `foldl'` is useful.

sdiehl commented:
This has only been in Data.List since 7.8 and we have to support everything back to 7.6. So the easiest way is to use the preprocessor to reimplement it for older versions and then after 7.6 just reexport it from base.

sdiehl commented on stack.yaml:
```diff
@@ -1,4 +1,4 @@
-resolver: lts-7.14
+resolver: lts-9.20
```
We leave this at the lowest LTS bound supporting GHC `7.6` and rely on testing to check for incompatibility with the later LTS resolvers. https://github.com/sdiehl/protolude/blob/master/all_stack.sh

cocreature commented:
I’ve added an implementation for `7.8` and removed the commit bumping the stackage snapshot (although I have to admit I’m not sure what the benefits of sticking to a snapshot which corresponds to `GHC 8.0.1` are). I haven’t included the rewrite rules that base includes for `scanl'`. If you want I can add those as well.

sdiehl commented:
Good to go. Cheers!
